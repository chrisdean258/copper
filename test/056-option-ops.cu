s = if true 1 else null
n = if false 1 else null

print(s || 2) # 1
print(n || 2) # 2
print(s || n) # 1
print(n || s) # 1
print(n && s) # null
print(s && n) # null
print(n ^^ s) # 1
print(s ^^ n) # 1
print(s ^^ s) # null
print(n ^^ n) # null

print(s == 1) # true
print(s == 2) # false
print(s == null) # false

print(n == 1) # false
print(n == 2) # false
print(n == null) # true
