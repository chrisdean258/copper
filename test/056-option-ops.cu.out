1
2
1
1
null
null
1
1
null
null
true
false
false
false
false
true
